### PR TITLE
Adds a windows pre compile/run hook to ensure an arch is specified

### PR DESCRIPTION
**Pull request by daserge**
Check that an actual architecture is specified for Windows 10 compile/run to avoid a runtime error

### CODE REVIEW

File: hooks/windows/check-arch.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+module.exports = function(ctx) {
+    if (ctx.opts && ctx.opts.platforms && ctx.opts.platforms.indexOf('windows') > -1
+        && ctx.opts.options) {
+        var shell = ctx.requireCordovaModule('shelljs');
+        var path = ctx.requireCordovaModule('path');
```
Comment (vladimir-kotikov) [inline hooks/windows/check-arch.js]: nit: `path` is not a Cordova module/dependency - you can use `require` here

File: hooks/windows/check-arch.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            'packageThumbprint': String,
+            'publisherId': String,
+            'buildConfig': String
+        }, {}, ctx.opts.options.argv, 0);
+
+        // Check if --appx flag is passed so that we have a project build version override:  
+        var isWin10 = args.appx && args.appx.toLowerCase() === 'uap';
+
+        // Else check "windows-target-version" preference:
+        if (!isWin10) {
```
Comment (vladimir-kotikov) [inline hooks/windows/check-arch.js]: Do we have a problems w/ AnyCpu architecture on Windows 8.1/Windows Phone 8.1? Shouldn't we just `return` here?

File: hooks/windows/check-arch.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            var configXml = shell.ls(path.join(ctx.opts.projectRoot, 'config.xml'))[0];
+
+            var reTargetVersion = /<preference\s+name="windows-target-version"\s+value="(.+)"\s*\/>/i;
```
Comment (vladimir-kotikov) [inline hooks/windows/check-arch.js]: I'd rather use `ConfigParser` from `common` here

File: hooks/windows/check-arch.js (hunk 2 above)
Comment (vladimir-kotikov) [inline hooks/windows/check-arch.js]: Ok, I get the idea. Please disregard the comment above

File: hooks/windows/check-arch.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        // Non-AnyCPU arch is required for Windows 10 (UWP) projects only:
+        if (isWin10) {
+            var rawArchs = ctx.opts.options.archs || args.archs;
+            var archs = rawArchs ? rawArchs.split(' ') : [];
+            // Filter out "anycpu"
+            archs = archs.filter(function (item) {
+                return item !== 'anycpu';
+            });
+
+            if (archs.length === 0) {
```
Comment (vladimir-kotikov) [inline hooks/windows/check-arch.js]: Should we also throw if there are multiple architectures specified _including_ AnyCpu? E.g. `['x86', 'x64', 'anycpu']`

Comment (daserge) [inline hooks/windows/check-arch.js]: Good point, thanks! I will update this.

**Pull request merged**